cachito/web/models.py

```diff
@@ -186,6 +187,32 @@ def to_purl(self):
             # see https://github.com/package-url/purl-spec/issues/63 for further reference
             purl_name = self.name.replace("/", "%2F")
             return f"pkg:golang/{purl_name}@{self.version}"
+
+        elif self.type == "npm":
+            purl_name = self.name.replace("@", "%40")
+            match = re.match(r"(?P<protocol>[^:]+):(?://)?(?P<suffix>.+)", self.version)
+            if not match:
+                return f"pkg:npm/{purl_name}@{self.version}"
+            protocol = match.group("protocol")
+            suffix = match.group("suffix")
+            if protocol in ("github", "gitlab", "bitbucket"):
+                # github:namespace/name#ref or gitlab:ns1/ns2/name#ref
+                match_forge = re.match(
+                    r"(?P<namespace>.+)/(?P<name>[^#/]+)#(?P<version>.+)$", suffix
+                )
+                if not match_forge:
+                    raise ValueError(f"Could not convert version {self.version} to purl")
```

Comment:
@athos-ribeiro and @lcarva I've been thinking about this... do you think a message should be exposed to the user rather than just a generic "Internal Server Error"? This would then be recorded by OSBS and it would make it easier for us to debug later on.

We could create a new exception that is called "CachitoContentManifestInvalid" and register it with Flask here:
https://github.com/release-engineering/cachito/blob/5d5ce769021101a2e7308cc35de01d9775a28193/cachito/web/app.py#L99-L102

Reply:
I like the idea. I did it in a separate commit.

Reply:
Nice, this new commits looks great!